magellan-library/src/main/java/com/wealthfront/magellan/lifecycle/LifecycleRegistry.kt

```diff
@@ -6,11 +6,11 @@ import com.wealthfront.magellan.lifecycle.LifecycleLimit.DESTROYED
 import com.wealthfront.magellan.lifecycle.LifecycleLimit.NO_LIMIT
 import com.wealthfront.magellan.lifecycle.LifecycleLimit.SHOWN
 
-internal class LifecycleRegistry : LifecycleAware {
+public class LifecycleRegistry : LifecycleAware {
```

Comment:
had to make this public to match LifecycleLimiter. IDK we are worried about exposing this

Reply:
I think if we're going to expose this already (which seems reasonable to me), we should just get rid of `LifecycleLimiter` altogether and just use `LifecycleRegistry` directly. Any methods that only exist in `LifecycleLimiter` now can either move into `LifecycleLimiter` directly or be extension functions on it following [extension oriented design](https://elizarov.medium.com/extension-oriented-design-13f4f27deaee)

Reply:
Good idea. I ended up merging them under `LifecycleRegistry`